src/ray/object_manager/ownership_based_object_directory.h

```diff
@@ -95,6 +95,7 @@ class OwnershipBasedObjectDirectory : public IObjectDirectory {
     // The node id that spills the object to the disk.
     // It will be Nil if it uses a distributed external storage.
     NodeID spilled_node_id = NodeID::Nil();
+    bool pending_creation = true;
```

Comment:
Actually, this is the fix. Turns out the pull manager assumes that the first location update will have the object size set. This was causing a bug where we would set the object size to 0 and try to activate the request.